llvm-general/Setup.hs

```diff
@@ -97,7 +97,12 @@ addLLVMToLdLibraryPath configFlags = do
   llvmConfig <- getLLVMConfig configFlags
   [libDir] <- liftM lines $ llvmConfig "--libdir"
   addToLdLibraryPath libDir
-                           
+
+-- | These flags are not relevant for us and dropping them allows
+-- linking against LLVM build with Clang using GCC
+ignoredCxxFlags :: [String]
+ignoredCxxFlags = ["-Wcovered-switch-default"]
```

Comment:
Please merge the pre-existing uncheckedHsFFIDefines into your new more general ignoredCxxFlags (and move the relevant parts of the comments on the individual list members).